dnx-on-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,263 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "virtualMachines_dnx_on_ubuntu_adminPassword": {
+      "defaultValue": "PASSw0rd",
+      "type": "securestring",
+      "metadata": {
+        "description": "The admin password for the Ubuntu machine"
+      }
+    },
+    "virtualMachines_dnx_on_ubuntu_adminusername": {
+      "defaultValue": "serveradmin",
+      "type": "string",
+      "metadata": {
+        "description": "The admin username for the Ubuntu machine"
+      }      
+    },
+    "virtualMachines_dnx_on_ubuntu_name": {
+      "defaultValue": "dnx-on-ubuntu",
+      "type": "string",
+      "metadata": {
+        "description": "The VM name"
+      }      
+    },
+    "networkInterfaces_dnx_on_ubuntu620_name": {
```

Comment:
NIC, NSG, VNET names aren't very critical to get a user input and can instead become variables. See #2 in best practices https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md